src/Isolate/UnitOfWork/Entity/ClassDefinition.php

```diff
@@ -77,7 +77,7 @@ public function getIdDefinition()
      */
     public function fitsFor($entity)
     {
-        return $this->className->isClassOf($entity);
+        return $this->className->isInstanceOf($entity);
```

Comment:
Actually this should be isClassOf becuase you check if a specific ClassName object represents class of specific entity. Entity::isInstanceOf(ClassName) could be the oposite check but we can't force any entity interface